tests/lib/metrics_test.py

```diff
@@ -0,0 +1,144 @@
+from datetime import datetime
+from unittest.mock import MagicMock, call, patch
+
+import boto3
+from cloudformation_cli_python_lib.interface import MetricTypes, StandardUnit
+from cloudformation_cli_python_lib.metrics import MetricPublisher, format_dimensions
+
+from botocore.stub import Stubber
+
+
+class MockSession:
+    def __init__(self, client):
+        self._client = client
+
+    def client(self, _name):
+        return self._client
+
+
+class MockLog:
+    def __init__(self, handler):
+        self.debug = handler
+
+
+def test_format_dimensions():
+    dimensions = {"MyDimensionKey": "val_1", "MyDimensionKey2": "val_2"}
+    result = format_dimensions(dimensions)
+    assert [
+        {"Name": "MyDimensionKey", "Value": "val_1"},
+        {"Name": "MyDimensionKey2", "Value": "val_2"},
+    ] == result
+
+
+@patch("cloudformation_cli_python_lib.metrics.LOG", auto_spec=True)
+def test_put_metric_catches_error(mock_logger):
+    client = boto3.client("cloudwatch")
+    stubber = Stubber(client)
+
+    stubber.add_client_error("put_metric_data", "InternalServiceError")
+    stubber.activate()
+
+    publisher = MetricPublisher("fake-namespace", MockSession(client))
+    fake_datetime = datetime(2019, 1, 1)
+    publisher.publish_exception_metric(fake_datetime, "CREATE", "fake-error")
+    stubber.deactivate()
+    expected_calls = [
+        call.error(
+            "An error occurred while publishing metrics: %s",
+            "An error occurred (InternalServiceError) when calling the PutMetricData operation: ",  # noqa: B950 # pylint: disable=line-too-long
```

Comment:
can split that string into 2 lines to avoid the lint excludes

```
            "An error occurred while publishing metrics: %s",
            "An error occurred (InternalServiceError) when calling the "
            "PutMetricData operation: ",
```

Reply:
nice tip, updated